kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/Quantity.java

```diff
@@ -408,6 +408,18 @@ public Quantity subtract(Quantity y) {
     return op(y, BigDecimal::subtract);
   }
 
+  /**
+   * Multiplies the quantity by the specified scalar multiplicand.
+   *
+   * @param multiplicand the scalar value to multiply by
+   * @return a new Quantity resulting from the multiplication of this quantity by the scalar multiplicand
+   */
+  public Quantity multiply(int multiplicand) {
+    BigDecimal numericalAmount = getNumericalAmount();
+    numericalAmount = numericalAmount.multiply(BigDecimal.valueOf(multiplicand));
+    return fromNumericalAmount(numericalAmount, format);
+  }
```

Comment:
Looks inconsistent with the rest of the operation methods defined in the class (namely add and subtract).

Wouldn't a `public Quantity multiply(Quantity y);` suffice for the described needs?

Reply:
In the Go client reviewers suggested that multiplicand should be a scalar rather than a Quantity. I'm not sure how to handle cases like `1G * 1n`, what should be the format of the result? Besides that, I don't think there are any practical use cases for Quantity by Quantity multiplications. When a multiplicand is a scalar, one the possible use cases may be to calculate total memory requests of a deployment, which would be a pod memory request multiplied by the replica count.

Reply:
> I don't think there are any practical use cases for Quantity by Quantity multiplications. When a multiplicand is a scalar, one the possible use cases may be to calculate total memory requests of a deployment, which would be a pod memory request multiplied by the replica count.

Agreed, this makes sense from a semantic perspective.

> In the Go client reviewers suggested that multiplicand should be a scalar rather than a Quantity

For reference: https://github.com/kubernetes/kubernetes/pull/117411/files#diff-c3c7dcd71c7c415cbf21c3d732510021ca5a11b46af7b633673aab30c33c717d

> I'm not sure how to handle cases like 1G * 1n, what should be the format of the result?

This situation is mostly the same for addition and subtraction of completely different types(CPU, memory)

The current implementation **should** be capable of taking care of this.

```java
  public Quantity multiply(Quantity y) {
    return op(y, BigDecimal::multiply);
  }
```

Anyway, it really doesn't make sense to compute a Quantity*Quantity from the user's perspective.